### PR TITLE
Unintegrated metrics

### PR DESCRIPTION
**Pull request by mumichae**
Fixed snakemake structure to allow for running metrics on unintegrated data.

- The unintegrated adata was preprocessed with `runPP.py` prior to metric computation. This was not the case before.
- unintegrated metrics are done for each scaling, feature subset and scenario
- unintegrated metrics files separated from integrated metrics files  `{scenario}/metrics_unintegrated/`
- unintegrated metrics are added to the final `metrics.csv` file

### CODE REVIEW

**Comment by LuckyMD** [comment]
This was still WIP, right @mumichae ?

File: scripts/metrics.py [hunk 1]
```diff
@@ -136,6 +139,11 @@
         # legacy check
         if ('emb' in adata_int.uns) and (adata_int.uns['emb']):
             adata_int.obsm["X_emb"] = adata_int.obsm["X_pca"].copy()
+        # fix for unintegrated metrics
+        if "X_emb" not in adata_int.obsm:
```
Comment (LuckyMD) [inline scripts/metrics.py:143]: why is this necessary? Can't you just encode unintegrated not as an embedding methods, but as `type=full`?

File: scripts/metrics.py [hunk 2]
```diff
@@ -244,11 +252,11 @@
                               lisi_graph_= lisi_graph_,
                               trajectory_=trajectory_
                              )
-    results.rename(columns={results.columns[0]:out_prefix}, inplace=True)
+    results.rename(columns={results.columns[0]:setup}, inplace=True)
     if verbose:
         print(results)
     # save metrics' results
-    results.to_csv(os.path.join(args.output, f'{out_prefix}.csv'))
+    results.to_csv(args.output)
```
Comment (LuckyMD) [inline scripts/metrics.py:259]: why do you no longer need the filename here?

File: scIB/preprocessing.py [hunk 3]
```diff
@@ -339,17 +339,26 @@ def hvg_batch(adata, batch_key=None, target_genes=2000, flavor='cell_ranger', n_
...
     checkAdata(adata)
     if batch_key:
         checkBatch(batch_key, adata.obs)
     
-    hvg = n_top_genes is not None
-    if hvg:
+    if n_top_genes is not None and ignore_hvg:
         print("HVG")
+        
+        ignore_hvg = False
```
Comment (LuckyMD) [inline scIB/preprocessing.py:360]: why is this needed?

File: config.yaml [hunk 4]
```diff
@@ -4,6 +4,8 @@ py_env : scIB-python
 conv_env : scIB-R-conversion
 timing: false
 
+unintegrated_metrics: false
```
Comment (LuckyMD) [inline config.yaml:7]: this is to toggle whether to run unintegrated metrics?

File: scripts/metrics.py (hunk 2 above)
Comment (mumichae) [reply scripts/metrics.py:259]: script now expects filename as input instead of directory. Should be the same as for any other rule

File: config.yaml (hunk 4 above)
Comment (mumichae) [reply config.yaml:7]: yes

Comment (mumichae) [reply config.yaml:7]: if not specified, will be interpreted as `False`
